core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -149,18 +149,11 @@ fn consume_scan_should_process_packet(
         return ProcessingDecision::Now;
     }
 
-    // Before sanitization, let's quickly check the static keys (performance optimization)
-    let message = &packet.transaction().get_message().message;
-    if !payload.account_locks.check_static_account_locks(message) {
-        return ProcessingDecision::Later;
```

Comment:
Assuming removing this doesn't have meaningful impact on performance? In any case, all for removing additional steps.

Reply:
Yeah, didn't notice any impactful difference when testing.
But moreso removing it because we need to load ALTs regardless of the outcome if this check now.

Previously we just skipped it and marked it for `Later`.
However, now we want to add those account locks so another lower-priority tx doesn't take them.
So if this check fails, we need to load ALTs to figure out and take the locks here anyway.